### PR TITLE
Add staking mode knob to harmony & node.sh

### PR DESCRIPTION
**Pull request by harmony-ek**
## Issue

This adds support for automatic shard ID calculation for staking mode in the `harmony` binary, and also adds the corresponding support (`-z` and `-Z` flags) to `node.sh`.

### Unit Test Coverage

Before:

```
ok      github.com/harmony-one/harmony/api/service/networkinfo  2.405s  coverage: 25.0% of statements
ok      github.com/harmony-one/harmony/cmd/client/wallet        0.262s  coverage: 3.8% of statements
ok      github.com/harmony-one/harmony/internal/configs/node    0.342s  coverage: 32.9% of statements
ok      github.com/harmony-one/harmony/internal/configs/sharding        0.356s  coverage: 21.4% of statements
ok      github.com/harmony-one/harmony/internal/utils   3.176s  coverage: 62.9% of statements
ok      github.com/harmony-one/harmony/node     2.104s  coverage: 14.2% of statements
```

After:

```
ok      github.com/harmony-one/harmony/api/service/networkinfo  2.302s  coverage: 25.0% of statements
ok      github.com/harmony-one/harmony/cmd/client/wallet        0.292s  coverage: 3.8% of statements
ok      github.com/harmony-one/harmony/internal/configs/node    0.358s  coverage: 39.8% of statements
ok      github.com/harmony-one/harmony/internal/configs/sharding        (cached)        coverage: 21.4% of statements
ok      github.com/harmony-one/harmony/internal/utils   (cached)        coverage: 58.9% of statements
ok      github.com/harmony-one/harmony/internal/wallet  0.456s  coverage: 77.8% of statements
ok      github.com/harmony-one/harmony/node     2.126s  coverage: 14.2% of statements
ok      github.com/harmony-one/harmony/p2p/utils        0.588s  coverage: 89.3% of statements
```

### Test/Run Logs

By default, legacy mode is enabled—note the lack of `-staking` option—and the given BLS key is looked up in the genesis table:
```
$ bin/harmony -blskey_file=e7f54994bc5c02edeeb178ce2d34db276a893bab5c59ac3d7eb9f077c893f9e31171de6236ba0e21be415d8631e45b91.key -blspass=pass: -network_type=testnet
My Genesis Account: one1yc06ghr2p8xnl2380kpfayweguuhxdtupkhqzw/e7f54994bc5c02edeeb178ce2d34db276a893bab5c59ac3d7eb9f077c893f9e31171de6236ba0e21be415d8631e45b91:0
Legacy mode; node key e7f54994bc5c02edeeb178ce2d34db276a893bab5c59ac3d7eb9f077c893f9e31171de6236ba0e21be415d8631e45b91 -> shard 0
^C
```

If, in legacy mode, the key is not found in the table (the example below looks for a testnet key in mainnet table), the node exits with an error message:

```
$ bin/harmony -blskey_file=e7f54994bc5c02edeeb178ce2d34db276a893bab5c59ac3d7eb9f077c893f9e31171de6236ba0e21be415d8631e45b91.key -blspass=pass: -staking=false -network_type=mainnet
cannot set up node account: cannot find key e7f54994bc5c02edeeb178ce2d34db276a893bab5c59ac3d7eb9f077c893f9e31171de6236ba0e21be415d8631e45b91 in table
```

Staking mode (`-staking=true`) uses the given BLS public key to deduce the shard ID.  Example for testnet (3 shards):
```
$ bin/harmony -blskey_file=95ea9dac42558d1252e6c01a16483783825f9c8a3044a6b35391343dea7ef2684ada3554ab3ef2cb495ed401fd728398.key -blspass=pass: -staking=true -network_type=testnet
Staking mode; node key 95ea9dac42558d1252e6c01a16483783825f9c8a3044a6b35391343dea7ef2684ada3554ab3ef2cb495ed401fd728398 -> shard 2
^C
```

Same key, for devnet (2 shards):

```
$ bin/harmony -blskey_file=95ea9dac42558d1252e6c01a16483783825f9c8a3044a6b35391343dea7ef2684ada3554ab3ef2cb495ed401fd728398.key -blspass=pass: -staking=true -network_type=devnet
Staking mode; node key 95ea9dac42558d1252e6c01a16483783825f9c8a3044a6b35391343dea7ef2684ada3554ab3ef2cb495ed401fd728398 -> shard 0
^C
```

## Operational Checklist

1. **Does this PR introduce backward-incompatible changes to the on-disk data structure and/or the over-the-wire protocol?**. (If no, skip to question 8.)

    **NO**

8. **Does this PR introduce backward-incompatible changes *NOT* related to on-disk data structure and/or over-the-wire protocol?** (If no, continue to question 11.)

    **YES**

9. **Does the existing `node.sh` continue to work with this change?**

    Yes – the new `-staking` flag is optional, and defaults to `false` (legacy mode).

10. **What should node operators know about this change?**

    Existing node operators who have been manually specifying shard ID (including P-OPS) should stop doing so and start using the new knobs.

    The node no longer falls back to staking mode upon genesis lookup failure by default; staking nodes must now be run with the `-staking=true` flag.

    * `node.sh` as of this version remains backward-compatible with older node by specifying `-staking` flag (not understood by older `harmony` binaries) only if `-z` flag is specified to enable staking.
    * 7 days after the staking epoch, `node.sh` will drop this backward-compatible behavior and always supply `-staking=false` or `-staking=true`: **Node operators on manual upgrade schedule will have to upgrade their `harmony` binary in order to use new `node.sh` after this change.**
    * 14 days after the staking epoch, the default `-staking` option will switch from `false` to `true`.  **Harmony-operated nodes running `harmony` binary directly will have to be run with `-staking=false` after this change.**
    * 21 days after the staking epoch, `node.sh` will default to `-z` (staking).  **Harmony-operated nodes running `node.sh` will have to be run with `-Z` after this change.**

    Since legacy mode needs to be available for Harmony-operated nodes, the `-Z`/`-z` flag of `node.sh` and the `-staking` flag of `harmony` will stay for a while.

11. **Does this PR introduce significant changes to the operational requirements of the node software, such as >20% increase in CPU, memory, and/or disk usage?**

    **NO**

### CODE REVIEW

File: scripts/node.sh [hunk 1]
```diff
@@ -116,6 +116,8 @@ usage: ${progname} [-1ch] [-k KEYFILE]
    -P             enable public rpc end point (default:off)
    -v             print out the version of the node.sh
    -V             print out the version of the Harmony binary
+   -z             run in staking mode
+   -Z             run in legacy, foundational-node mode (default)
```
Comment (rlan35) [inline scripts/node.sh]: change to other char.

Comment (harmony-ek) [reply scripts/node.sh]: Done.

File: cmd/harmony/main.go [hunk 2]
```diff
@@ -447,7 +453,20 @@ func main() {
 
 	if *nodeType == "validator" {
 		setupInitialAccount()
```
Comment (rlan35) [inline cmd/harmony/main.go]: Also disable the part for legacy account lookup when it's staking.

Comment (harmony-ek) [reply cmd/harmony/main.go]: Separated legacy/staking processing.

**Pull request merged**